inference-engine/src/transformations/src/transformations/convert_batch_to_space.cpp

```diff
@@ -7,40 +7,119 @@
 #include <memory>
 #include <vector>
 
-#include <ngraph/opsets/opset2.hpp>
+#include <ngraph/builder/reshape.hpp>
+#include <ngraph/opsets/opset3.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph/rt_info.hpp>
 
 void ngraph::pass::ConvertBatchToSpace::convert_batch_to_space() {
-    auto input0 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input1 = ngraph::op::Constant::create(element::i64, Shape{4}, Shape{1, 1, 1, 1});
-    auto input2 = ngraph::op::Constant::create(element::i64, Shape{4}, {0, 0, 0, 0});
-    auto input3 = ngraph::op::Constant::create(element::i64, Shape{4}, {0, 0, 0, 0});
-    auto batch_to_space = std::make_shared<ngraph::opset2::BatchToSpace>(input0, input1, input2, input3);
-
+    auto batch_to_space = ngraph::pattern::wrap_type<ngraph::opset3::BatchToSpace>();
     ngraph::graph_rewrite_callback callback = [](pattern::Matcher& m) {
-        auto batch_to_space = std::dynamic_pointer_cast<ngraph::opset2::BatchToSpace> (m.get_match_root());
+        auto batch_to_space = std::dynamic_pointer_cast<ngraph::opset3::BatchToSpace> (m.get_match_root());
         if (!batch_to_space) {
             return false;
         }
-        auto last_node = batch_to_space->decompose_op()[0];
-        last_node.get_node()->set_friendly_name(batch_to_space->get_friendly_name());
-        ngraph::replace_node(batch_to_space, last_node.get_node_shared_ptr());
+
+        auto data = batch_to_space->input_value(0);
+        auto block = batch_to_space->input_value(1);
+        auto crops_begin = batch_to_space->input_value(2);
+        auto crops_end = batch_to_space->input_value(3);
+
+        const auto& data_shape = data.get_shape();
+
+        const auto block_const = std::dynamic_pointer_cast<opset3::Constant>(block.get_node_shared_ptr());
+        const auto crops_begin_const = std::dynamic_pointer_cast<opset3::Constant>(crops_begin.get_node_shared_ptr());
+        const auto crops_end_const = std::dynamic_pointer_cast<opset3::Constant>(crops_end.get_node_shared_ptr());
+
+        if (!block_const || !crops_begin_const || !crops_end_const) {
+            return false;
+        }
+
+        std::vector<int64_t> block_values, crops_end_values;
+        block_values = block_const->cast_vector<int64_t>();
```

Comment:
You make two copies here.
1. cast_vector creates a copy
2. vector copy constructor